python/cuml/ensemble/randomforestregressor.pyx

```diff
@@ -531,8 +529,8 @@ class RandomForestRegressor(Base):
         if n_cols != self.n_cols:
             raise ValueError("The number of columns/features in the training"
                              " and test data should be the same ")
-        preds = np.zeros(n_rows,
-                         dtype=self.dtype)
+        preds = cuda.device_array(n_rows,
+                                  dtype=self.dtype)
         cdef uintptr_t preds_ptr
         preds_m, preds_ptr, _, _, _ = \
```

Comment:
Why do we have to do input_to_dev_array if we already have a cuda_device array? Couldn't we just use the allocated array directly?

Reply:
i can replace `input_to_dev_array` with `get_dev_array_ptr` function but we need to use one of the two functions as cython does not let us pass the address of a python variable